### ISSUE TITLE
`TypeScript declaration files should never be required`

### ISSUE THREAD

**Issue by tycho01** [s-panferov/awesome-typescript-loader#422]
In my project I'm using some of Angular's internal classes. As the JS part of Angular gets bundled, whats left in place of the original `.ts` source in its npm module are `.d.ts` typings containing the definitions I'd like to annotate my code with.

This repo doesn't seem to like this though, erroring with `TypeScript declaration files should never be required`.
I'd understand that just requiring/importing a whole TS declaration file seems silly. However, I'm under the impression that my use-case of plucking a type from it to annotate my code with (as TS expects me to) is legitimate.
Am I missing something here? Or is grabbing types from repos that only put the non-`.d.ts` parts in bundles not supported?

**Comment by s-panferov** [comment]
You should never see the case in normal workflow. The error means that you're trying to require a module-based .d.ts file using `require` calls.

**Comment by tycho01** [comment]
@s-panferov:
I believe the specific syntax used was `import` rather than `require`, but I presume that was beside your point.

I presume the part about angular distributing a pre compiled version is considered normal.
So then, I suppose wanting to import a type from a file (all 
`.d.ts` of course) not exposed through their main barrel is not?

Less common, sure, but I'd think the use case is legitimate...

**Comment by s-panferov** [comment]
TypeScript should elide all plain type imports from a source code, this should not happen. If you're still struggling with the error and still interested — please file a repro.

**Comment by d-ph** [comment]
Hi @s-panferov ,

I'm getting the `TypeScript declaration files should never be required` when overriding typings distributed with an npm package as described in [this stackoverflow answer](https://stackoverflow.com/questions/41558661/ignore-bundled-d-ts-and-use-external-declarations/41559327#41559327). It essentially involves using the following piece of `tsconfig.json` config:
```json
"compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "validate.js":  ["./validate.d.ts"]
    }
  }
```
When I run `tsc` manually, it correctly outputs a
```
var aLib = require("aLib");
```
and my overridden typings are being picked up by `tsc`, but when I run my webpack build, the `atl.TsConfigPathsPlugin` seems to be feeding webpack with the definition file instead of the js file, when `aLib` is being `import`ed:
```javascript
/***/ 46:
/***/ (function(module, exports) {

/***/ }),
```

Would it be possible for `atl.TsConfigPathsPlugin` not to rewrite `import` paths, if the `tsconfig.json` is remapping a library path to a `*.d.ts` file, please?

**Comment by pearldiver** [comment]
I got the same problem when i add below code:
import { TestBed } from '@angular/core/testing';
in my spec-bundle.ts

**Comment by s-panferov** [comment]
Should be fixed in v3.2.0

**Comment by thomascmost** [comment]
Hey, I'm seeing this with two node_modules, moment and symbol-observable... I'm running at-loader v.3.2.3 -- any ideas?
